Tests/NIOTests/EventLoopTest.swift

```diff
@@ -91,6 +91,89 @@ public class EventLoopTest : XCTestCase {
+            if counter.load() == 0 {
+                XCTAssertTrue(DispatchTime.now().uptimeNanoseconds - nanos >= initialDelay.nanoseconds)
+            } else if counter.load() == count {
+                expect.fulfill()
+                repeatedTask?.cancel()
+            }
+            counter.store(counter.load() + 1)
+        }
+
+        waitForExpectations(timeout: 0.07) { _ in
+            XCTAssertEqual(counter.load(), count + 1
+            )
+            XCTAssertTrue(DispatchTime.now().uptimeNanoseconds - nanos >= initialDelay.nanoseconds + count * delay.nanoseconds)
+        }
+    }
+
+    public func testScheduleRepeatedTaskCancelFromDifferentThread() throws {
+        let nanos = DispatchTime.now().uptimeNanoseconds
+        let initialDelay: TimeAmount = .milliseconds(5)
+        let delay: TimeAmount = .milliseconds(10)
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer {
+            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
+        }
+
+        let expect = expectation(description: "Is cancelling RepatedTask")
+        var repeatedTask: RepeatedTask?
+        repeatedTask = eventLoopGroup.next().scheduleRepeatedTask(initialDelay: initialDelay, delay: delay) { () -> Void in
+            DispatchQueue.main.async {
+                repeatedTask?.cancel()
+                expect.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: 0.02) { _ in
+            XCTAssertTrue(DispatchTime.now().uptimeNanoseconds - nanos >= initialDelay.nanoseconds)
+        }
+    }
+
+    public func testScheduleRepeatedTaskToNotRetainRepeatedTask() throws {
+        let initialDelay: TimeAmount = .milliseconds(5)
+        let delay: TimeAmount = .milliseconds(10)
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+
+        var repeated: RepeatedTask?
+        weak var weakRepeated: RepeatedTask?
+        repeated = eventLoopGroup.next().scheduleRepeatedTask(initialDelay: initialDelay, delay: delay) { () -> Void in }
+        weakRepeated = repeated
+        XCTAssertNotNil(weakRepeated)
+        repeated?.cancel()
+        XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
+        repeated = nil
+        XCTAssertNil(weakRepeated)
+    }
+
+    public func testScheduleRepeatedTaskToNotRetainEventLoop() throws {
+        let initialDelay: TimeAmount = .milliseconds(5)
+        let delay: TimeAmount = .milliseconds(10)
+        var eventLoopGroup: EventLoopGroup? = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        weak var weakEventLoop = eventLoopGroup?.next()
+
+        eventLoopGroup?.next().scheduleRepeatedTask(initialDelay: initialDelay, delay: delay) { () -> Void in }
+        XCTAssertNoThrow(try eventLoopGroup?.syncShutdownGracefully())
+        XCTAssertNotNil(weakEventLoop)
+        eventLoopGroup = nil
+        Thread.sleep(forTimeInterval: 0.01)
```

Comment:
I had to introduce a sleep here otherwise this test was really flaky and sometimes the `weakEventLoop` wasn't nil. Do you have any idea why that could be the case?